integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/action/imagecatalog/ImageCatalogGetImagesByNameAction.java

```diff
@@ -17,29 +20,47 @@
 
     private CloudPlatform platform = CloudPlatform.MOCK;
 
+    private String stackName;
+
     public ImageCatalogGetImagesByNameAction() {
     }
 
     public ImageCatalogGetImagesByNameAction(CloudPlatform platform) {
         this.platform = platform;
     }
 
+    public ImageCatalogGetImagesByNameAction(String stackName) {
+        this.stackName = stackName;
+    }
+
     @Override
     public ImageCatalogDto action(TestContext testContext, ImageCatalogDto entity, CloudbreakClient cloudbreakClient) throws Exception {
         LOGGER.info("Get images of ImageCatalog within workspace by catalog name: {}", entity.getRequest().getName());
         try {
-            entity.setResponseByProvider(
-                    cloudbreakClient
-                            .getCloudbreakClient()
-                            .imageCatalogV4Endpoint()
-                            .getImagesByName(cloudbreakClient.getWorkspaceId(), entity.getName(), null, platform.name())
-            );
+            ImageCatalogV4Endpoint imageCatalogV4Endpoint = cloudbreakClient
+                    .getCloudbreakClient()
+                    .imageCatalogV4Endpoint();
+
+            entity.setResponseByProvider(getImagesV4Response(entity, cloudbreakClient, imageCatalogV4Endpoint));
             logJSON(LOGGER, "images have been fetched successfully: ", entity.getRequest());
         } catch (Exception e) {
             LOGGER.warn("Cannot get images of ImageCatalog : {}", entity.getRequest().getName());
             throw e;
         }
         return entity;
     }
+
+    private ImagesV4Response getImagesV4Response(ImageCatalogDto entity, CloudbreakClient cloudbreakClient, ImageCatalogV4Endpoint imageCatalogV4Endpoint)
+            throws Exception {
+        ImagesV4Response imagesByName;
```

Comment:
Personally i prefer
```
return StringUtils.isNoneEmpty(stackName)
            ? imageCatalogV4Endpoint
                     .getImagesByName(cloudbreakClient.getWorkspaceId(), entity.getName(), stackName, null)
            : imageCatalogV4Endpoint
                     .getImagesByName(cloudbreakClient.getWorkspaceId(), entity.getName(), null, platform.name());
```

Reply:
Thanks, it looks better this time, will change it if stg needs to be modified around there. :)